### PR TITLE
Sync with catcoder

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java [hunk 1]
```diff
@@ -1,11 +1,56 @@
...
+            user.addInvitedChallenge(challenge);
+            userRepository.save(user);
+        }
+    }
 
+    private User mapParticipationDto(ParticipationDto dto){
+        User user = new User();
+        user.setUsername(dto.getName());
+        // NOTE is it the same encryption?
+        user.setPassword(dto.getPassword());
```
Comment (Exordian) [inline platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java]: it is

File: platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java [hunk 2]
```diff
@@ -1,11 +1,56 @@
...
+        for(ParticipationDto participation: dto.getParticipations()){
+            User user = mapParticipationDto(participation);
+            user.addInvitedChallenge(challenge);
+            userRepository.save(user);
+        }
+    }
 
+    private User mapParticipationDto(ParticipationDto dto){
+        User user = new User();
+        user.setUsername(dto.getName());
```
Comment (Exordian) [inline platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java]: that will fail the regex on username ; take a look on the lab instance

File: platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java [hunk 3]
```diff
@@ -1,11 +1,56 @@
...
+        this.challengeRepository = challengeRepository;
+    }
 
-    public void updateUserFromCodingcontest(ParticipationDto dto) {
+    public User updateUserFromCodingcontest(ParticipationDto dto) {
+        return userRepository.save(mapParticipationDto(dto));
+    }
+
+    public void createContest(CodingcontestDto dto){
+        Challenge challenge = new Challenge();
```
Comment (Exordian) [inline platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java]: you should not create a new challenge everytime the button gets hit, this is also used for updating the contest

Comment (Exordian) [inline platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java]: so look first if the contest already exists and update it

File: platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java [hunk 4]
```diff
@@ -1,11 +1,68 @@
...
+        if(user==null){
+            user = mapParticipationDto(dto);
+        }else{
+            user.setPassword(dto.getPassword());
+        }
+        return userRepository.save(user);
+    }
+
+    public void createContest(CodingcontestDto dto) {
+        String canonicalName = "ccc-" + dto.getLocation() + "-" + dto.getGameName();
```
Comment (Exordian) [inline platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java]: i'd use the received uuid for the canonicalName, the game switches too often afterwards ^^

Comment (victorbalan) [inline platform-codingcontest-sync/src/main/java/uno/cod/platform/server/codingcontest/sync/service/CodingcontestSyncService.java]: the canonical name should be something readable..that`s why i went with this approach.

**Comment by Exordian** [comment]
partly merged, codingcontest => platform sync working

**Comment by flowlo** [comment]
So you mean this PR is obsolete as of 2b0bedc02f412461da12df44129dc895d1b40983? If so, please close it. Otherwise: What part was "merged", what still needs to be?

**Pull request closed without merging**